### DISCUSSION TITLE
The Basic Idea

### DISCUSSION THREAD

**Discussion by momiller121** [momiller121/westjet-weather#2] [Ideas]
The purpose of this project is to create a space for the purpose of learning. 

The basic idea is that we want to create a new, simple API resource that is the result of combining two other upstream public API resources. Further, we aim to explore test driven development, behaviour driven development and the use of GitHub discussions and issues as the basis of distributed communication. 

This API product will be called 'westjet-weather'. It is, simply, a GET resource that uses the WestJet Destination Service API and a public weather API to produce a response that enhances the WestJet airport information with realtime weather data.

- https://api.westjet.com/destination-service/v1/en-CA/airports/{iataCode}
- https://www.metaweather.com/api/ (we will use this lightly to respect the owner's resources)

The sole parameter to the API is a path parameter called iataCode. This is a three letter (all capital letters) code.

There should be strict input validation on the format of the `iataCode` parameter. It must be three letters, all caps.

If the WestJet Destination API returns a 404 for an iataCode, so should the `westjet-weather` resource. 

If the WestJet Destination API returns a non-200 code (except for 404) the `westjet-weather` resource should return a 500 error.

If the WestJet Destination API returns 200 but there is a problem retrieving the weather, the `westjet-weather` resource should return the airport info and indicate in the payload that the weather info was not available at this time.

**Comment by Basharfrancis** [comment]
Thank you Mark!

**Comment by momiller121** [comment]
The main 'logic' involved in this effort is in the resolution of the location identifier for the weather API by using the latitude and longitude provided in the airport payload from the WestJet API. It will require a location search on the weather API (to get the `woeid`) followed by a request for the weather by that `woeid`.
